res_handler_jq_transform.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/mitchellh/mapstructure"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+type ResponseTransformJQMiddleware struct {
+	Spec *APISpec
+}
+
+func (h *ResponseTransformJQMiddleware) Init(c interface{}, spec *APISpec) error {
+	h.Spec = spec
+
+	return nil
+}
+
+func (h *ResponseTransformJQMiddleware) HandleResponse(rw http.ResponseWriter, res *http.Response, req *http.Request, ses *SessionState) error {
+	_, versionPaths, _, _ := h.Spec.Version(req)
+	found, meta := h.Spec.CheckSpecMatchesStatus(req, versionPaths, TransformedJQResponse)
+	if !found {
+		return nil
+	}
+
+	defer res.Body.Close()
+
+	t := meta.(*TransformJQSpec)
+
+	var bodyObj interface{}
+	dec := json.NewDecoder(res.Body)
+	if err := dec.Decode(&bodyObj); err != nil {
+		return err
+	}
+	jqObj := map[string]interface{}{
+		"body":       bodyObj,
+		"reqContext": ctxGetData(req),
```

Comment:
should be called `_tyk_context` to be similar to follow same guidelines https://tyk.io/docs/concepts/context-variables/